src/partition/InputLines.cc

```diff
@@ -279,14 +315,17 @@ CharPtrPair const InputLines::Impl::read(char *buf) {
 
 // Method delegation.
 
-InputLines::InputLines(std::vector<fs::path> const &paths, size_t blockSize, bool skipFirstLine)
-        : _impl(boost::make_shared<Impl>(paths, blockSize, skipFirstLine)) {}
+InputLines::InputLines(std::vector<fs::path> const &paths, size_t blockSize, bool skipFirstLine,
+		       std::vector<std::string> paramNames)
+        : _impl(boost::make_shared<Impl>(paths, blockSize, skipFirstLine, paramNames)) {}
 
 size_t InputLines::getBlockSize() const { return _impl ? _impl->getBlockSize() : 0; }
 
 size_t InputLines::getMinimumBufferCapacity() const { return _impl ? _impl->getMinimumBufferCapacity() : 0; }
 
 bool InputLines::empty() const { return _impl ? _impl->empty() : true; }
+  
+std::vector<std::string> InputLines::getParameterNames() const { return _impl->getParameterNames(); }
```

Comment:
(for efficiency) Should return the `const` reference:
```
std::vector<std::string> const& InputLines::getParameterNames() const { ... }
```

Reply:
This function is not needed anymore